calliope/core/preprocess/locations.py

```diff
@@ -389,8 +391,14 @@ def process_per_distance_constraints(tech_name, tech_settings, locations, locati
     return tech_settings
 
 
-def check_costs_and_compute_depreciation_rates(tech_id, loc_or_link, tech_config, warnings, errors):
+def check_costs_and_compute_depreciation_rates(tech_id, loc_or_link, tech_config, warnings, errors, run_mode, parent):
     cost_classes = list(tech_config.get('costs', {}).keys())
+    # Adds spores_score details if "spores" run mode is selected
+    if run_mode == 'spores' and parent != 'demand': # not in str(tech_id): # not working: the idea is to find a way to skip demand techs, by checking the parent tech in tech_config, if possible
```

Comment:
I would try to avoid anything to do with run mode here. The point of 'model' and 'run' being separate is that anything to do with the run mode should not influence anything until a user runs `model.run()`. It's worth thinking about how that influences what is done in this file.